net.solarnetwork.node.weather.ibm.wc/src/net/solarnetwork/node/weather/ibm/wc/BasicWCClient.java

```diff
@@ -77,14 +83,32 @@ private BigDecimal parseBigDecimal(JsonNode n) {
 		if ( n.isNull() ) {
 			return null;
 		}
-		return new BigDecimal(n.asLong());
+		return n.decimalValue();
 	}
 
 	private LocalTime parseTimeValue(JsonNode n) {
 		if ( n == null || n.isNull() ) {
 			return null;
 		}
-		return new LocalTime(n.asLong() * 1000);
+		try {
+			return new LocalTime(this.tsFormat.parse(n.asText()));
+		} catch ( ParseException e ) {
+			return null;
+		}
+
+	}
+
+	private Date parseDateValue(JsonNode n) {
```

Comment:
If you make this method return a `DateTime` object instead of a `Date` you'll more easily be able to convert to `LocalDate` and `LocalTime` objects.